Testing/Dashboard/Telemetry/TelemetryTests.cs

```diff
@@ -0,0 +1,192 @@
+using Dashboard;
+using Dashboard.Server.Persistence;
+using Dashboard.Server.Telemetry;
+using Content;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Testing.Dashboard.Telemetry
+{
+    public class TelemetryNunitTests
+    {
+        [Test]
+        public void OnAnalyticsChanged_GetUserCountVsTimeStamp_UpdatingUserCount()
+        {
+            // Arrange
+            UserData user1= new UserData("Noah",1);
+            SessionData session1= new SessionData();
+            session1.AddUser(user1);
+            UserData user2= new UserData("Shailee",2);
+            SessionData session2=new SessionData();
+            session2.AddUser(user1);
+            session2.AddUser(user2);
+            //Act
+            DateTime currTime= new DateTime(2021,11,23,1,0,0);
+            TelemetryFactory.GetTelemetryInstance().OnAnalyticsChanged(session1,currTime);
+            int userCount1 = TelemetryFactory.GetTelemetryInstance().userCountAtEachTimeStamp[currTime];
+            bool check1=false;
+            // total users in session1 is 1
+            if(userCount1==1) check1=true;
```

Comment:
Please add Assert.AreEquals(session1.users.count, userCount1) rather than adding a check variable

Reply:
Earlier I did the same but it was giving errors.

Reply:
Oh okay I will try it after merging